build.sbt

```diff
@@ -1670,6 +1680,19 @@ lazy val annotationProcSetting = Seq(
     )
 )
 
+lazy val javadocSettings = Seq(
+  Compile / doc / javacOptions --= Seq(
+    "-deprecation",
+    "-g",
+    "-Xlint:unchecked",
+    "-proc:full"
+  ),
+  Compile / doc / javacOptions ++= Seq(
```

Comment:
Regular `javacOptions` depending on some path in tests? That is an odd choice

Reply:
First of all, these options are for `javadoc`, not `javac` compilation itself.

The dependency is because of `--snippet-path`. Since [JEP-413](https://openjdk.org/jeps/413) there is a support for including pieces of code in generated Javadoc pages. The [Persistance API is using that](https://github.com/enso-org/enso/blob/b5969cf31c8b33a2095b9212a334acae927bc2a6/lib/java/persistance/src/main/java/org/enso/persist/Persistance.java#L15) and is referencing the [snippet in test directory](https://github.com/enso-org/enso/blob/b5969cf31c8b33a2095b9212a334acae927bc2a6/lib/java/persistance/src/test/java/org/enso/persist/PersistanceTest.java#L18). Hence it needs path to `src/test/java`. 

That's all desirable - by referencing code that is executed as part of tests, we guarantee the code is compilable and functional.

Reply:
OK, fine. Thanks for the explanation.